week3/homework/src/index.js

```diff
@@ -3,80 +3,81 @@
 const express = require('express');
 const Joi = require('joi');
 const app = express();
-const todos = require('./todos');
+const validateTodos = require('./validateTodos');
+const todo = require('./todo.json');
 app.use(express.json());
 
-// get the list of todos
-app.get('/todos', (req, res) => {
-  res.send(todos);
+// get the list of todo
+app.get('/todo', (req, res) => {
+  res.send(todo);
 });
 
 // get todo by id number
-app.get('/todos/:id', (req, res) => {
-  const todo = todos.find(c => c.id === parseInt(req.params.id));
-  if (!todo) return res.status(404).send('the given api not found');
-  res.send(todo);
+app.get('/todo/:id', (req, res) => {
+  const todoItem = todo.find(i => i.id === parseInt(req.params.id));
+  if (!todoItem) return res.status(404).send('Id not found');
+  res.send(todoItem);
 });
 
 // add todo
-app.post('/todos', (req, res) => {
+app.post('/todo', (req, res) => {
   const { error } = validateTodos(req.body);
   if (error) {
     return res.status(400).send(error.details[0].message);
   }
 
-  const todo = {
-    id: todos.length + 1,
-    name: req.body.name,
+  const todoItem = {
+    id: todo.length + 1,
+    name: req.body.description,
     done: false,
   };
-  todos.push(todo);
-  res.send(todos);
+  todo.push(todoItem);
```

Comment:
This is not saved. You have to explicitly write this to the file system.

Reply:
Thanks i will update it